### PR TITLE
Add Router and dynamic URL

### LINKED ISSUE #4: Router
## Value statement 

As a **person**
I need **to share the result of my county to who ever i want**
so that **other people are easy informed**

## Description (Text, Scribble, Wireframe, Design)

Add router with unique URL paths for the county result page.

## Acceptance criteria

-   [x] If the URL of the result page is shared via copy paste, other user should get the same result page.
-   [x] If the user types a place in the URL the county will be searched and the result page for that county is called.

## Tasks

-   [x] Add react-router
-   [x] Add react-router dynamic URL
-   [x] Add loading logic
-   [x] Test on mobile and desktop

## Size

small

### PR DESCRIPTION
**Pull request by JulianBlohm**
Hello, I've just added react router dom and created a dynamic URL, so that the county the user searched for is always in the URL of the result page. Now the user can share the result via copy paste and can open a result page directly.
If the user types a random place in the URL, the geocoding will find the correct county and the county result is displayed.
Also the result page loads just if the data is available.

Thanks for your time 🦦

### CODE REVIEW

File: src/ResultPage.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+function ResultPage({setUserPlace, countyData}) {
 
     const [countyClassification, setCountyClassification] = useState('')
+    
+    let { id } = useParams()
 
     useEffect(() => classifyCountyIncidence(), [countyData])
-
+    useEffect(() => {!countyData.countyName && setUserPlace(id)}, [id])
+    console.log(id)
```
Comment (CinFrei) [inline src/ResultPage.js]: Found a console.log(counter=1).

File: src/hooks/useUserLocation.js [hunk 2]
```diff
@@ -16,11 +19,14 @@ export default function useUserLocation() {
     })
     const [ errorMessage, setErrorMessage ] = useState()
     const [ isCountyDataLoaded, setIsCountyDataLoaded] = useState(false)
-
+    console.log(countyData)
+    console.log(userPlace)
```
Comment (CinFrei) [inline src/hooks/useUserLocation.js]: Found another two console.log(counter=3).

File: src/components/Form.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+import PropTypes from 'prop-types'
 import Button from './Button'
 import Input from './Input'
-import PropTypes from 'prop-types'
-import styled from 'styled-components/macro'
-import { useState } from 'react'
 
 
-function Form({setUserPlace, errorMessage}) {
+function Form({setUserPlace, errorMessage,}) {
```
Comment (CinFrei) [inline src/components/Form.js]: A lone comma.

File: src/MainPage.js [hunk 4]
```diff
@@ -1,8 +1,11 @@
+import { useEffect } from 'react'
 import styled from 'styled-components/macro'
 import Form from './components/Form'
 
 
-function MainPage({setUserPlace, errorMessage}) {   
+function MainPage({setUserPlace, errorMessage, newSearch}) {   
+    
+    useEffect(() => newSearch(), [])
```
Comment (CinFrei) [inline src/MainPage.js]: NewSearch is a bit confusing, I would call it resetSearch.

**Review by CinFrei — COMMENTED** [review]
Nice work! :rocket:

File: src/hooks/useUserLocation.js [hunk 5]
```diff
@@ -1,8 +1,11 @@
 import { useState, useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
```
Comment (mmurakaru) [inline src/hooks/useUserLocation.js:2]: Good one ;)

File: src/hooks/useUserLocation.js [hunk 6]
```diff
@@ -50,11 +56,15 @@ export default function useUserLocation() {
         setIsCountyDataLoaded(false)
     }
 
+    function showResultPage() {
+        const countyNameCompact = countyData.countyName.replace(/ /g,'')
+        history.push(`/${countyNameCompact}`)
+    }
+
```
Comment (mmurakaru) [inline src/hooks/useUserLocation.js:62]: Awesome 😊

**Review by mmurakaru — APPROVED** [review]

File: src/hooks/useUserLocation.js [hunk 7]
```diff
@@ -16,11 +19,13 @@ export default function useUserLocation() {
     })
     const [ errorMessage, setErrorMessage ] = useState()
     const [ isCountyDataLoaded, setIsCountyDataLoaded] = useState(false)
-
+    
     useEffect(() => {userPlace && getCountyData()}, [userPlace])
     useEffect(() => {coordinates.longitude && getIncidenceData()}, [coordinates])
     useEffect(() => setErrorMessage(''), [userPlace,coordinates] )
```
Comment (mmurakaru) [inline src/hooks/useUserLocation.js:25]: Could you combine line 23 and 25 in one useEffect?

File: src/index.js [hunk 8]
```diff
@@ -1,12 +1,15 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App';
+import { BrowserRouter as Router } from "react-router-dom";
```
Comment (mmurakaru) [inline src/index.js:3]: Is renaming this necessary?

File: src/hooks/useUserLocation.js (hunk 7 above)
Comment (JulianBlohm) [reply src/hooks/useUserLocation.js:25]: Sadly not, cause line 23 should just happen if there is a change with userPlace, but the error message should also be reset after coordinates changed

File: src/index.js (hunk 8 above)
Comment (JulianBlohm) [reply src/index.js:3]: Not necessary but common and I like it short :)

**Review by FabianAlleblas — COMMENTED** [review]
Nice one 👍

**Pull request merged**